mantidproject/mantid
# Fixed bug in FitGaussianPeaksAlgorithm

**Description of work.**
Fixed issue with peak being at the end of range due to not enough data points being present to fit all parameters

**To test:**
1. Run code below
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

ws = CreateWorkspace([0,3,6,9,12,15,18,21,24,27,30,33]*2, [0,0,0,1,5,1,2,8,1,0,10,7]+[0]*12,NSpec=2)
table = CreateEmptyTableWorkspace()
table.addColumn("float", "Centre")
table.addRow([12])
table.addRow([21])
table.addRow([30])

FitGaussianPeaks(ws, table)
```
2. This should produce 4 tables and a workspace with 3 peaks
![image](https://user-images.githubusercontent.com/59264035/123423394-6aa71b80-d5b7-11eb-975d-17b4a7bdfa4e.png)

3. The table named peak_table there should be 2 entries  one peak with a centre of about 12 height of 5 , sigma of about 1.6 and area of about 21 and another peak with a centre of about 21 height of 8.1 , sigma of about 1.6 and area of about 39

4. The table named refit_peak_table should be empty

5. The table named fit_cost should have on row with the chi2 having a value of 1.418 and poisson -326.998.






Fixes #31879 .




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
